### ISSUE TITLE
what if I have multiple Turtle files I want to check?

### ISSUE THREAD

**Issue by egonw** [vemonet/jena-riot-action#1]
Of course, I can repeat the `- uses`, but can I give a wild card too, perhaps?

**Comment by vemonet** [comment]
The validation is done with Jena riot (in a docker container), so this will depend on this library

We can find more details in their documentation: https://jena.apache.org/documentation/io/

But it does not seems like it supports wildcard unfortunately

**Comment by stain** [comment]
`riot` will expect the shell to do the expansion; perhaps using `sh -c` could help. But on the other hand I also think you will struggle to define the action with variable number of arguments to add into `args`

**Comment by vemonet** [comment]
Indeed, my "implementation" of the action is the simplest possible with only an `action.yml`, so it 100% rely on how Jena RIOT Docker image works 

An option to improve it would be to extend the [Jena RIOT docker image](https://github.com/stain/jena-docker/blob/master/jena/Dockerfile) with an `entrypoint.sh` scripts that would be able to handle multiple files and more complex input arguments

A bit like I did for this Action: https://github.com/vemonet/sparql-operations-action


Another option is to create a matrix of the files you want to validate and execute Jena riot on each file  this dynamically generated matrix (could be also static, but that's not fun)

The advantage with this solution is that you will have a nice reporting of which file failed directly in the GitHub Actions workflow logs! The problem is that you will need to provide a static list of the files to process, or you will need to do some GitHub Actions magics to dynamically generate the matrix based on the files you want to process in a first job, and use this matrix in a 2nd job to validate each RDF file
